### PR TITLE
Fix markdownlint on all *.md files

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

- Add a local configuration to ignore more rules than `test-infra`
- Fix rules that are not ignored

The main reason is to fix them once and for all, so that new PR that changes `.md` files get validate only on what has been added

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- <del>[ ] Includes [tests](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md#principles) (if functionality changed/added)</del>
- [x] Includes [docs](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md#principles) (if user facing)
- <del>[ ] Commit messages follow [commit message best practices](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md#commit-messages)</del>

_See [the contribution guide](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md)
for more details._

cc @bobcatfish

### CODE REVIEW

**Comment by pivotal-nader-ziada** [comment]
cool pr, but can I ask how come you want to ignore the list bullets rules (MD004, MD005 and MD007)? is this too much to change them everywhere?

**Comment by vdemeester** [comment]
@pivotal-nader-ziada right, so the bullets rules are a pain, for example it doesn't allow to have unordered list nested in ordered one… The indentation one seems to also have false positive (as it fails on a list that has the correct level of indentation)…

File: test/markdown-lint-config.rc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# For help, see
+# https://github.com/markdownlint/markdownlint/blob/master/docs/configuration.md 
+
+# Ignore long lines
```
Comment (adrcunha) [inline test/markdown-lint-config.rc]: MD013 is about long lines. I suggest briefly describing what other rules are being ignored, so the next person updating this file doesn't need hunt them down in the documentation.

Comment (vdemeester) [reply test/markdown-lint-config.rc]: @adrcunha good point :+1:

**Pull request merged**